lib/RecommendationProvider.js

```diff
@@ -0,0 +1,157 @@
+  this._recommendedContent = [];
+  this._blockedRecommendedContent = [];
+  this._currentRecommendation = null;
+  this._pocketEndpoint = simplePrefs.prefs[POCKET_PREF];
+  this._previewProvider = previewProvider;
+  this._asyncExpireRecommendations(this.options.pocketTimeout);
+}
+
+RecommendationProvider.prototype = {
+  _pocketTimeoutID: null,
+
+  /**
+    * If we need a recommendation, go get one
+    */
+  getARecommendation(getRecommendation) {
+    if (getRecommendation) {
+      // if we are already displaying a recommendation, just show that one and don't get a new one
+      if (this._currentRecommendation) {
+        return this._currentRecommendation;
+      }
+      return this._getRandomRecommendation();
+    }
+    // if we don't need a recommendation, don't return one
+    return null;
+  },
+
+  /**
+    * Get a new recommendation from Pocket
+    */
+  getANewRecommendation(getRecommendation) {
+    this._currentRecommendation = null;
+    this.getARecommendation(getRecommendation);
+  },
+
+  /**
+    * Randomly pick a recommendation from the allowed set of recommendations (i.e ones that are not blocked)
+    */
+  _getRandomRecommendation() {
+    const allowedRecommendations = this._recommendedContent.filter(item => !this._blockedRecommendedContent.includes(item.url));
+    if (allowedRecommendations.length > 0) {
+      let index = Math.floor(Math.random() * (allowedRecommendations.length));
+      this._currentRecommendation = Object.assign({}, {
+        url: allowedRecommendations[index].url,
+        timestamp: allowedRecommendations[index].timestamp,
+        recommended: true});
+      return this._currentRecommendation;
+    }
+    this._currentRecommendation = null;
+    return this._currentRecommendation;
+  },
+
+  /**
+    * Fetches Pocket recommendations and saves them to a global list of recommended content
+    */
+  asyncSetRecommendedContent: Task.async(function*() {
+    try {
+      let response = yield this._asyncGetRecommendationsFromPocket();
+      if (response.ok) {
+        let responseJson = yield response.json();
+        if (responseJson.urls.length > 0) {
+          // wait for the recommendation's metadata to come back (we don't want to show ugly recommendations)
+          let richRecommendations = yield this._previewProvider.getLinkMetadata(responseJson.urls);
+          this._recommendedContent = richRecommendations;
+          this._recommendedContent.forEach(item => item.timestamp = this._convertTime(item.timestamp));
+        }
+      }
+    } catch (err) {
+      Cu.reportError(err);
+      throw err;
+    }
+  }),
+
+  /**
+    * Update the list of recommendations the user has blocked (they do not want to see these again)
+    */
+  setBlockedRecommendation(link) {
+    this._blockedRecommendedContent.push(link);
+    this._currentRecommendation = null;
+  },
+
+  /**
+    * Convert milliseconds into human readable form and style it accordingly
+    */
+  _convertTime(timestamp) {
```

Comment:
instead of humanizing the timestamps here, can we do them in-content using `moment` (which we're already using for the highlights cards)?

http://momentjs.com/docs/#/durations/humanize/